Packs/CheckPointHarmonyEndpoint/Integrations/CheckPointHarmonyEndpoint/CheckPointHarmonyEndpoint.py

```diff
@@ -0,0 +1,2703 @@
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        request_body["operationParameters"] |= {
+            "hive": args.get("hive"),
+            "key": args.get("key"),
+            "valueName": args.get("value_name"),
+            "isRedirected": arg_to_bool(args.get("is_redirected")),
+        }
+        response = client.agent_registry_key_delete(remove_empty_elements(request_body))
+        args["job_id"] = response.get("jobId")
+
+    return schedule_command(args, client, demisto.command())
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get("interval", 30)),
+    timeout=arg_to_number(demisto.args().get("timeout", 600)),
+    poll_message="File copy request is executing",
+    requires_polling_arg=False,
+)
+def agent_file_copy_command(args: Dict[str, Any], client: Client) -> PollResult:
+    """Copies the given file from the given source to the given destination on computers matching the given query.
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        request_body["operationParameters"] |= {
+            "sourceAbsolutePath": args.get("destination_absolute_path"),
+            "destinationAbsolutePath": args.get("source_absolute_path"),
+        }
+        response = client.agent_file_copy(remove_empty_elements(request_body))
+
+        args["job_id"] = response.get("jobId")
+
+    return schedule_command(args, client, demisto.command())
+
+
+@polling_function(
+    name=demisto.command(),
+    interval=arg_to_number(demisto.args().get("interval", 30)),
+    timeout=arg_to_number(demisto.args().get("timeout", 600)),
+    poll_message="File move request is executing",
+    requires_polling_arg=False,
+)
+def agent_file_move_command(args: Dict[str, Any], client: Client) -> PollResult:
+    """Moves the given file from the given source to the given destination on computers matching the given query.
+
+    Args:
+        client (Client): Harmony API client.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    if not args.get("job_id"):
+        request_body = build_request_body(args)
+        request_body["operationParameters"] |= {
+            "sourceAbsolutePath": args.get("destination_absolute_path"),
+            "destinationAbsolutePath": args.get("source_absolute_path"),
+        }
+        response = client.agent_file_move(remove_empty_elements(request_body))
```

Comment:
Why do we do the `remove_empty_elements` again? The arguments that you added are required.
Same question for when sending the `agent_file_delete` request.

Reply:
I removed the remove_empty_elements function in both commands.